src/meta/src/hummock/manager/mod.rs

```diff
@@ -3225,77 +3226,95 @@ impl HummockManager {
+                                }
+
+                                pick_task_count
+                            }
+
                             if let Some(compactor) =
                                 hummock_manager.compactor_manager.get_compactor(context_id)
                             {
                                 if let Some((group, task_type)) =
                                     hummock_manager.auto_pick_compaction_group_and_type().await
                                 {
-                                    let selector: &mut Box<dyn CompactionSelector> = {
-                                        let versioning_guard =
-                                            read_lock!(hummock_manager, versioning).await;
-                                        let versioning = versioning_guard.deref();
-
-                                        if versioning.write_limit.contains_key(&group) {
-                                            let enable_emergency_picker = match hummock_manager
-                                                .compaction_group_manager
-                                                .read()
-                                                .await
-                                                .try_get_compaction_group_config(group)
-                                            {
-                                                Some(config) => {
-                                                    config.compaction_config.enable_emergency_picker
-                                                }
-                                                None => {
-                                                    unreachable!("compaction-group {} not exist", group)
-                                                }
-                                            };
-
-                                            if enable_emergency_picker {
-                                                compaction_selectors
-                                                    .get_mut(&TaskType::Emergency)
-                                                    .unwrap()
-                                            } else {
-                                                compaction_selectors.get_mut(&task_type).unwrap()
-                                            }
-                                        } else {
-                                            compaction_selectors.get_mut(&task_type).unwrap()
+                                    let mut selector = compaction_selectors.get_mut(&task_type).unwrap();
+
+                                    let pick_task_count = pull_task_handler(hummock_manager.clone(), pull_task_count, group, selector, task_type, compactor.clone()).await;
+                                    // if the pick_task_count is less than `EXPECTED_PULL_TASK_RATIO`, try to pull the left tasks from emergency selector when write-stop
+                                    const EXPECTED_PULL_TASK_RATIO: f32 = 0.75;
```

Comment:
Same with the comment above. Will it be better to pass the `pick_task_count: &mut usize` for limitation across pulls instead of limiting each pull to 75%?